cmd/query/app/querysvc/query_service_v2.go

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package querysvc
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger/cmd/query/app/querysvc/adjuster"
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/iter"
+	"github.com/jaegertracing/jaeger/storage_v2/depstore"
+	"github.com/jaegertracing/jaeger/storage_v2/tracestore"
+)
+
+var errNoArchiveSpanStorage = errors.New("archive span storage was not configured")
+
+const (
+	defaultMaxClockSkewAdjust = time.Second
+)
+
+// QueryServiceOptions has optional members of QueryService
+type QueryServiceOptionsV2 struct {
+	ArchiveTraceReader tracestore.Reader
+	ArchiveTraceWriter tracestore.Writer
+	Adjuster           adjuster.Adjuster
+}
+
+// StorageCapabilities is a feature flag for query service
+type StorageCapabilitiesV2 struct {
+	ArchiveStorage bool `json:"archiveStorage"`
+	// TODO: Maybe add metrics Storage here
+	// SupportRegex     bool
+	// SupportTagFilter bool
+}
+
+// QueryService contains span utils required by the query-service.
+type QueryServiceV2 struct {
+	traceReader      tracestore.Reader
+	dependencyReader depstore.Reader
+	options          QueryServiceOptionsV2
+}
+
+// NewQueryService returns a new QueryService.
+func NewQueryServiceV2(
+	traceReader tracestore.Reader,
+	dependencyReader depstore.Reader,
+	options QueryServiceOptionsV2,
+) *QueryServiceV2 {
+	qsvc := &QueryServiceV2{
+		traceReader:      traceReader,
+		dependencyReader: dependencyReader,
+		options:          options,
+	}
+
+	if qsvc.options.Adjuster == nil {
+		qsvc.options.Adjuster = adjuster.Sequence(adjuster.StandardAdjusters(defaultMaxClockSkewAdjust)...)
+	}
+	return qsvc
+}
+
+// GetTrace is the queryService implementation of tracestore.Reader.GetTrace
+func (qs QueryServiceV2) GetTraces(ctx context.Context, traceIDs ...tracestore.GetTraceParams) iter.Seq2[[]ptrace.Traces, error] {
+	getTracesIter := qs.traceReader.GetTraces(ctx, traceIDs...)
+	return func(yield func([]ptrace.Traces, error) bool) {
+		foundTraceIDs := make(map[pcommon.TraceID]struct{})
+		getTracesIter(func(traces []ptrace.Traces, err error) bool {
+			if err != nil {
+				return yield(nil, err)
+			}
+			for _, trace := range traces {
+				resources := trace.ResourceSpans()
+				for i := 0; i < resources.Len(); i++ {
+					scopes := resources.At(i).ScopeSpans()
+					for j := 0; j < scopes.Len(); j++ {
+						spans := scopes.At(j).Spans()
+						for k := 0; k < spans.Len(); k++ {
+							span := spans.At(k)
+							foundTraceIDs[span.TraceID()] = struct{}{}
+						}
+					}
+				}
+			}
+			return yield(traces, nil)
+		})
+		if qs.options.ArchiveTraceReader != nil {
+			missingTraceIDs := []tracestore.GetTraceParams{}
```

Comment:
```suggestion
			var missingTraceIDs []tracestore.GetTraceParams
```
no need to init to empty slice